pkg/gofr/datasource/opentsdb/dropcache.go

```diff
@@ -0,0 +1,60 @@
+package opentsdb
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/otel/trace"
+)
+
+type DropCachesResponse struct {
+	StatusCode     int
+	DropCachesInfo map[string]string `json:"DropcachesInfo"`
+	logger         Logger
+	tracer         trace.Tracer
+	ctx            context.Context
+}
+
+func (dropResp *DropCachesResponse) SetStatus(code int) {
+	setStatus(dropResp.ctx, dropResp, code, "SetStatus-DropCaches", dropResp.logger)
+}
+
+func (dropResp *DropCachesResponse) setStatusCode(code int) {
+	dropResp.StatusCode = code
+}
+
+func (dropResp *DropCachesResponse) GetCustomParser() func(respCnt []byte) error {
+	return getCustomParser(dropResp.ctx, dropResp, "GetCustomParser-DropCaches", dropResp.logger,
+		func(resp []byte) error {
+			return json.Unmarshal([]byte(fmt.Sprintf(`{"DropcachesInfo":%s}`, string(resp))), &dropResp)
+		})
+}
```

Comment:
I'm not convinced by this pattern

Please consider this

```suggestion
func (dropResp *DropCachesResponse) GetCustomParser() func(respCnt []byte) error {
	return getCustomParser(dropResp.ctx, dropResp, "GetCustomParser-DropCaches", dropResp.logger,
		func(resp []byte) error {
			var j map[string]string
			err := json.Unmarshal(resp, &j)
			if err != nil {
				return err
			}
			dropResp.DropCachesInfo = j
			return nil
		})
}
```